x-pack/plugins/canvas/canvas_plugin_src/functions/common/alterColumn.ts

```diff
@@ -52,23 +52,25 @@ export function alterColumn(): ExpressionFunctionDefinition<
         return input;
       }
 
-      const column = input.columns.find((col) => col.name === args.column);
+      const column = input.columns.find(
```

Comment:
This is going to find the first column that name or id matches.  

Let's change it so that we are consistent.  Look for ID.  If not an ID match, look for name.  

Example (this sets the name of the first column to the id of the second column.  From there, using the 2nd column id will match the first column name instead of the 2nd column id): 
```
esaggs index={indexPatternLoad id="90943e30-9a47-11e8-b64d-95841ca0b247"} 
  aggs={aggHistogram id="a" enabled=true schema="segment" field="bytes" maxBars=34 interval="auto" has_extended_bounds=false min_doc_count=false}
  aggs={aggSum id="b" enabled=true schema="metric" field="bytes"} metricsAtAllLevels=false partialRows=false
| head 1
| alterColumn column="col-0-a" type="number" name="col-1-b"
| alterColumn column="col-1-b" name="Altered"
```

Reply:
Fixed.